canonical-web-and-design/maas-ui
# Refactor HeaderStrip. Implement update selection functionality in machine list.

## Done
- Split HeaderStrip into smaller components because it was starting to get large. Now it consists of:
  - AddHardwareMenu
  - TakeActionMenu
  - HeaderStripTabs
  - ActionFormWrapper (which is bare bones right now, but will house the individual form components for each action type) 
- Built the "update your selection" functionality from the angular machine list, which filters selected machines to only those that can perform the selected action.

## QA
- Select a bunch of machines.
- Open the take action menu and pick an action that not all machines can perform.
- Check that a warning shows up that asks you to update your selection.
- Click the "update your selection" link and check that the selected machines are filtered to those that can perform the action.
- Check that the submit label is formatted nicely for each action.

## Fixes
Fixes canonical-web-and-design/MAAS-squad#1793

## Screenshots
### Select action that can't be performed by all selected machines
![0 0 0 0_8400_MAAS_r_machines (14)](https://user-images.githubusercontent.com/25733845/74955980-b6671500-5405-11ea-8898-73d5ecedfca5.png)

### Error message
![0 0 0 0_8400_MAAS_r_machines (15)](https://user-images.githubusercontent.com/25733845/74955987-b9fa9c00-5405-11ea-8426-2c3ec28d268a.png)

### All machines can perform action
![0 0 0 0_8400_MAAS_r_machines (16)](https://user-images.githubusercontent.com/25733845/74955993-bc5cf600-5405-11ea-9f87-cecc0449047c.png)
